Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntryUtils.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntryUtils.h>
+
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzCore/Serialization/Utils.h>
+#include <AzCore/IO/GenericStreams.h>
+#include <AzCore/IO/FileIO.h>
+
+#include <QMimeData>
+#include <QUrl>
+#include <AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/SourceAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/ProductAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/FolderAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/RootAssetBrowserEntry.h>
+
+namespace AzToolsFramework
+{
+    namespace AssetBrowser
+    {
+        namespace Utils
+        {
+            void ToMimeData(QMimeData* mimeData, const AZStd::vector<const AssetBrowserEntry*>& entries)
+            {
+                if (!mimeData)
+                {
+                    return;
+                }
+                if (entries.empty())
+                {
+                    return;
+                }
+
+                AZStd::string outputString = ToString(entries);
+
+                // capture the URLs (full paths to items) in case you're about to drop this mimedata
+                // onto something which does not understand "asset browser entry" but does understand URLs.
+                QList<QUrl> urls;
```

Comment:
Thew way Mimedata works is that the sender (ie, whatever gui widget **begins** the drag and drop operator) attaches multiple different kinds of data to the same mimedata object. 
For example, when you copy (or drag) text from Visual STudio, VS adds both "text/plain" AND "text/rich" datas - one without any markup, and one with syntax highlight and color.

Its up to the recieving application to inspect the mimedata and see if there's anything of use in it.  if you drop the text into WORD or EXCHANGE for example, its going to see that text/rich is available and use that.  If you drop it into notepad, it ignores the text/rich data, and sees if text/plain is there.

Proper encoding of mimedata means offering the right amount of information to potential drop targets.  

If we _dont_ include the URLS in the mimedata then applications that can eat URLS may not be able to use it.

Example
drag and drop the asset browser entry into various apps:

- Into Visual Studio Code, with URLS: Code prefers URLS -  opens the actual asset file itself in a new tab.  So for example if its a PNG file, it will actually show you the image of the PNG file.
- Into Visual STudio Code, with only text/plain: Code doesn't find any URLs, doesn't know how to handle Asset Browser Entries, and so pastes the _text_ into your current open document (so you see' Source: {some guid} // filename' which could be useful
- Into MSPaint:  MSpaint will open the actual image if it happens to be a URL of a file it can eat.  otherwise wont do anything.  Without URLS, it refuses all drops.

So basically, it depends on what information you'd like to supply to other applications to make use of.  If you want them to understand that this represents files on disk, you include URLS.  If you don't want them to understand this, then you don't include URLS.  URLS are pretty universal, and almost all applications that accept files understand URLs.